File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.js
function solution(new_id) {
    const stepOneTwo = new_id.toLowerCase().match(/[a-z0-9-_.]/g).join("");
    const stepThree = stepOneTwo.replace(/\.+/g, ".");
    const stepFour = stepThree.replace(/^\.|\.$/g, "");
    const stepFive = stepFour.length === 0 ? "a" : stepFour;
    const stepSix = stepFive.length >= 16 ? stepFive.slice(0, 15).replace(/\.$/g, "") : stepFive;
    if(stepSix.length <= 2){
        const len = stepSix.length;
        const addChar = stepSix[len-1].repeat(3-len);
        const stepSeven = stepSix + addChar;
        return stepSeven;
    }
    return stepSix;
}

File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.py
import re

def solution(new_id):
    step_one_two = ''.join(re.findall(r'[a-z0-9\-_.]', new_id.lower()))
    step_three = re.sub(r'\.+', '.', step_one_two)
    step_four = step_three.strip('.')
    step_five = 'a' if len(step_four) == 0 else step_four
    step_six = step_five[:15].rstrip('.') if len(step_five) >= 16 else step_five
    if len(step_six) <= 2:
        step_six += step_six[-1] * (3 - len(step_six))
    
    return step_six
